### PR TITLE
122 create a semanticsviewer component

### PR DESCRIPTION
**Pull request by Vismayak**

### CODE REVIEW

File: src/reducers/semantics.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+	loading: boolean
+}
+
+const defaultState = {semantics: [], loading: false};
+
+const semantics = (state:SemanticState = defaultState, action:SemanticAction) => {
+	switch(action.type) {
+	case RECEIVE_SEMANTICS:
+		return Object.assign({}, state, {semantics: action.semantics});
+	case DELETE_ITEM:
```
Comment (longshuicy) [inline src/reducers/semantics.js:26]: there is no delete?

File: src/utils/flowtype.js [hunk 2]
```diff
@@ -64,6 +64,53 @@ export type FileDescriptor = {
...
+		columns: SemanticColumn[];
+	}
+};
+
+export type Semantics = Semantic[];
+
+export type SemanticState = {
+	semantics: Semantics;
+};
+
```
Comment (longshuicy) [inline src/utils/flowtype.js:113]: we are not using flow to check types anymore ( i think )... so next time we can try remove this file and see what happens

**Comment by longshuicy** [comment]
image

Don't you want to prettify this section based on the feedback you gathered yesterday?

File: src/reducers/semantics.js (hunk 1 above)
Comment (Vismayak) [reply src/reducers/semantics.js:26]: I thought delete functionality was to be handled by us and not the users?

Comment (longshuicy) [reply src/reducers/semantics.js:26]: Then you don't need to add to reducer? :-)

File: src/utils/flowtype.js [hunk 3]
```diff
@@ -64,6 +64,53 @@ export type FileDescriptor = {
...
+	"xsd": string;
+	"qudt": string;
+	"unit": string;
+	"openvocab": string;
+
+}
+
+export type SemanticColumn = {
+	name: string;
+	titles: string;
```
Comment (ylyangtw) [inline src/utils/flowtype.js:86]: maybe use `Title`? so it would be more consistent with other headers

**Comment by ylyangtw** [comment]
I have a question and this may be another potential issue -- do we want to add space information?

File: src/utils/flowtype.js (hunk 3 above)
Comment (ylyangtw) [reply src/utils/flowtype.js:86]: and maybe should do align left?

Comment (ylyangtw) [reply src/utils/flowtype.js:86]: image

Comment (Vismayak) [reply src/utils/flowtype.js:86]: Hmm it is aligned for me

Comment (Vismayak) [reply src/utils/flowtype.js:86]: Anyways Chen mentioned we will remove the file after this release

File: src/components/SemanticViewer.jsx [hunk 4]
```diff
@@ -0,0 +1,533 @@
...
+				previewLoading: false,
+				semanticJSON: jsonData
+			});
+		} catch (error) {
+			console.error("Error fetching JSON data:", error);
+			this.setState({
+				previewLoading: false
+			});
+		}
+	};
```
[the remark this answers is not in the document]
Comment (Rashmil-1999) [reply src/components/SemanticViewer.jsx:283]: refactor it to a functional component?

**Comment by Rashmil-1999** [comment]
> if you click next page and then click on some type name, the page seems to be reset to 1...

I don't find this issue, could you post a screenshot of it?

File: src/components/SemanticViewer.jsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (Rashmil-1999) [reply src/components/SemanticViewer.jsx:283]: Ohh, I see now... we need to convert it to a Functional component :p we are again using class-based approach for this component.

**Review by ylyangtw — APPROVED** [review]

**Review by Rashmil-1999 — APPROVED** [review]

**Pull request merged**